src/state_machine.zig

```diff
@@ -2471,6 +2554,21 @@ pub fn StateMachineType(
     };
 }
 
+// TODO(client_release_min): When client_release_min is bumped, remove this function and the
+// legacy code it gates.
+//
+// Specifically, when forbid_zero_amounts() is true:
+// - Zero-amount transfers are forbidden (`amount_must_not_be_zero`).
+// - Post-pending-transfer uses amount=0 as a sentinel for "post full amount".
+// - Balancing transfers use amount=0 as a sentinel for `maxInt(u128)`.
+fn forbid_zero_amounts(client_release: vsr.Release) bool {
```

Comment:
Suggestion:
I think we could build a pattern for future cases and scope this kind of function into a namespace (or at least some common prefix).

Example:
`compatibility.forbid_zero_amounts(...)`
`legacy.allow_zero_amounts(...)`

Reply:
I don't think it needs to be a namespace -- these functions would only ever live in `state_machine.zig`.
Though, if we end up adding more than one we can reconsider.